commodore/dependency_mgmt/__init__.py

```diff
@@ -211,9 +216,8 @@ def register_components(cfg: Config):
 
         c = registered_components[cn]
         aspec = cspecs[alias]
-        if aspec.url != c.repo_url or aspec.path != c.sub_path:
-            raise NotImplementedError("Changing alias sub path / URL NYI")
-        c.register_alias(alias, aspec.version)
+        if aspec.url == c.repo_url and aspec.path == c.sub_path:
+            c.register_alias(alias, aspec.version)
```

Comment:
This function is used when running Commodore in "local mode" (`--local`). From my quick local test this will now fail with a less comprehensible error than the original draft's `NotImplementedError` since we still don't handle changed URL/path here.

We probably want something like

```suggestion
        if aspec.url != c.repo_url:
            adep = cfg.register_dependency_repo(aspec.url)
            adep.register_component(alias, component_dir(cfg.work_dir, alias))
        c.register_alias(alias, aspec.version, aspec.path)
```